### PR TITLE
Add new hook `update_plugins_secondary_variables_extra`

### PR DESCRIPTION
**Pull request by ggrbill**

### CODE REVIEW

File: alfasim_sdk/hook_specs.py [hunk 1]
```diff
@@ -146,6 +146,18 @@ def update_plugins_secondary_variables(ctx: "void*") -> "int":
...
 
+def update_plugins_secondary_variables_extra(ctx: "void*") -> "int":
+    """
+    Internal simulator hook to update plugin's secondary variables in the ExtraVarSolver scope.
+    ExtraVarsolver is used to solve the tracer transport equation.
+    This is called as the last step on ALFAsim's ExtraVarSolver update variables workflow.
+
+    :param ctx:
+
+    :returns: Return OK if successful or anything different if failed
```
Comment (williamjamir) [inline alfasim_sdk/hook_specs.py]: This OK it's a macro?

File: alfasim_sdk/hook_specs.py [hunk 2]
```diff
@@ -146,6 +146,18 @@ def update_plugins_secondary_variables(ctx: "void*") -> "int":
     """
 
 
+def update_plugins_secondary_variables_extra(ctx: "void*") -> "int":
+    """
+    Internal simulator hook to update plugin's secondary variables in the ExtraVarSolver scope.
+    ExtraVarsolver is used to solve the tracer transport equation.
```
Comment (williamjamir) [inline alfasim_sdk/hook_specs.py]: `ExtraVarSolver`

File: alfasim_sdk/hook_specs.py [hunk 3]
```diff
@@ -146,6 +146,18 @@ def update_plugins_secondary_variables(ctx: "void*") -> "int":
     """
 
 
+def update_plugins_secondary_variables_extra(ctx: "void*") -> "int":
+    """
+    Internal simulator hook to update plugin's secondary variables in the ExtraVarSolver scope.
```
Comment (williamjamir) [inline alfasim_sdk/hook_specs.py]: Just a opinion, you can used it or not 😋 
```
Hook used internally by the simulator to update the plugins secondary variables on the ExtraVarSolver scope.

This hook will be executed right after the routine ExtraVarSolver has being finished, as the last step before continuing with the normal flow.

File: alfasim_sdk/hook_specs.py [hunk 4]
```diff
@@ -146,6 +146,18 @@ def update_plugins_secondary_variables(ctx: "void*") -> "int":
     """
 
 
+def update_plugins_secondary_variables_extra(ctx: "void*") -> "int":
```
Comment (williamjamir) [inline alfasim_sdk/hook_specs.py]: This hook's only used to update variables related with "tracers". Hence, I think that this hook should have a more meaningful name mentioning at least the tracers on his name.

Furthermore, we already have a hook named `update plugins secondary variables`, and adding a new hook just adding an `_extra` at the end could be confusing forcing the developer to explicitly look at the documentation to get the difference.

File: alfasim_sdk/hook_specs.py (hunk 1 above)
Comment (ggrbill) [reply alfasim_sdk/hook_specs.py]: It is an `enum` from `alfasim-sdk\alfasim_sdk_api\common.h`

File: alfasim_sdk/hook_specs.py (hunk 3 above)
Comment (ggrbill) [reply alfasim_sdk/hook_specs.py]: The second line is not what happens.

File: alfasim_sdk/hook_specs.py (hunk 4 above)
Comment (ggrbill) [reply alfasim_sdk/hook_specs.py]: But it is not only for tracers. The `ExtraVarSolver` can be used to solve extra variables, for instance, the moments from PBE of hydrates plugin.

**Pull request merged**